### DISCUSSION TITLE
Environment configuration pre-requisites?

### DISCUSSION THREAD

**Discussion by reubster** [AltimateAI/vscode-dbt-power-user#44] [Q&A] [answered]
Hi All, 

I use vscode;
- on Windows 10 
- Python 3.8.3 
- dbt 0.18.1. I 
- pipenv to create virtualenv's
- pwsh (7.0) as my terminal 
- snowflake 

I have dbt Power User installed but quite a few the features don't work. 

What does work; 
- Run current model 
- terminal runs dbt list. 
- parent / child model navigation 
- install / update dbt 
- log tailing 

What does not work; 
- no compiled SQL launch
- no hyperlink go to definition 
- autocompletion 

Can't find any errors anywhere, and can't see why this behaviour might be occurring. 

Any ideas? 

Thanks,

**Comment by reubster** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/19515061/104585118-7e8b4080-565b-11eb-8e21-68f64a60f3c5.png)

**Reply by reubster** [reply]
Actually, the hyperlink to definition does work. You have to hold down ctrl on a windows machine.

**Reply by reubster** [reply]
also I now have intellisense working... 

does the extension need dbt_project to be in the _workspace_ root ?

**Reply by mdesmet** [reply]
All DBT projects in every workspace folder will be discovered automatically and it will provide specific autocompletes for each project.

**Reply by jomach** [reply]
how does the discovery work ? I have my project under src/project1/dbt_project.yml

**Comment by datanerd09** [comment]
Hi All,

I use VSCode.

- dbt version: 1.0.8
- python version: 3.9.12
- os info: macOS-11.6.2-x86_64-i386-64bit
- pipenv to create virtualenv's
- zsh as my terminal
- snowflake

I have dbt Power User installed but quite a few the features don't work.
I can't get it to do any of the following:

- Run current model
- terminal runs dbt list.
- parent / child model navigation
- no compiled SQL launch
- no hyperlink go to definition
- autocompletion

At startup, the terminal displays the following (truncating path to the project directory):
```sh
An error occurred while trying to refresh the project configuration: Error: ENOENT: no such file or directory, open '../zion-data-modeling/dbt_packages/dbt-expectations-0.5.5/dbt_project.yml'
An error occurred while trying to refresh the project configuration: Error: ENOENT: no such file or directory, open '../zion-data-modeling/dbt_packages/dbt-expectations-0.5.5/integration_tests/dbt_project.yml'
An error occurred while trying to refresh the project configuration: Error: ENOENT: no such file or directory, open '../zion-data-modeling/dbt_packages/dbt-date-0.5.7/dbt_project.yml'
An error occurred while trying to refresh the project configuration: Error: ENOENT: no such file or directory, open '../zion-data-modeling/dbt_packages/dbt-date-0.5.7/integration_tests/dbt_project.yml'
An error occurred while trying to refresh the project configuration: Error: ENOENT: no such file or directory, open '../zion-data-modeling/dbt_packages/dbt-utils-0.8.6/dbt_project.yml'
An error occurred while trying to refresh the project configuration: Error: ENOENT: no such file or directory, open '../zion-data-modeling/dbt_packages/dbt-utils-0.8.6/integration_tests/dbt_project.yml'
```

The `dbt_expectations` package is in the root of the project, but not with the names inclusive of version as alluded to with the ENOENT error:
<img width="199" alt="image" src="https://user-images.githubusercontent.com/94391358/186527119-8d6773f8-fe90-46c7-bfa2-51ae4a037949.png">
But, these get installed via the `dbt_expectations` package in my `packages.yml` in the root of the project:
<img width="286" alt="image" src="https://user-images.githubusercontent.com/94391358/186527508-e93fedb0-da8f-4cb8-a635-daaa1c0557b9.png">

Any thoughts on what is going on here and how I might get around this issue?
Appreciate any help with this in advance!

**Reply by mdesmet** [reply]
interesting, not sure how that can happen.

I will convert this into a bug.
